MovieChecker/MovieChecker/BoxOfficeViewController.m

```diff
@@ -0,0 +1,148 @@
+        self.session = [NSURLSession sessionWithConfiguration:config];
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    [self moviesOnRottenTomatoes];
+}
+
+- (void)moviesOnRottenTomatoes
+{
+    NSURL *url = [RottenTomatoes boxOfficeURL];
+    NSURLSessionDataTask *dataTask = [self.session dataTaskWithURL:url completionHandler:^(NSData *data,
+                                                                                           NSURLResponse *response,
+                                                                                           NSError *error) {
+        if (!error) {
+            NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+            if (httpResp.statusCode == 200) {
+                [self interpretBoxOfficeMoviesFromArray:[RottenTomatoes arrayFromData:data withKey:@"movies"]];
+            } else {
+                // TODO: Handle bad status code!
+            }
+        } else {
+            // TODO: Handle the error!
+        }
+    }];
+    [dataTask resume];
+}
+
+- (void)interpretBoxOfficeMoviesFromArray:(NSArray *)boxOfficeArray
+{
+    [self.movies removeAllObjects];
+    Movie *movie;
+    for (NSDictionary *movieData in boxOfficeArray) {
+        // TODO: Fix placeholders
+        
+        movie = [[Movie alloc] initWithTitle:movieData[@"title"]
+                                   imagePath:movieData[@"posters"][@"thumbnail"]
+                                     urlPath:movieData[@"links"][@"self"]];
+        movie.rating = movieData[@"mpaa_rating"];
+        movie.synopsis = movieData[@"synopsis"];
+        movie.year = [movieData[@"year"] intValue];
+        movie.runtime = [movieData[@"runtime"] intValue];
+        [self.movies addObject:movie];
+    }
+    [self.tableView reloadData];
+}
+
+- (void)imageForMovie:(Movie *)movie forCell:(BoxOfficeCell *)cell atIndexPath:(NSIndexPath *)indexPath
+{
+    NSURL *url = [NSURL URLWithString:movie.imagePath];
+    NSURLSessionDataTask *downloadImageTask = [[NSURLSession sharedSession]
+                                                   dataTaskWithURL:url
+                                                   completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                                       if (!error) {
+                                                           NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+                                                           if (httpResp.statusCode == 200) {
+                                                               UIImage *downloadedImage = [[UIImage alloc] initWithData:data];
+                                                               //DLOG(@"");
+                                                               movie.image = downloadedImage;
+                                                               dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
It's slightly preferable to use `[[NSOperationQueue mainQueue] addOperationWithBlock:^{ ... }];`, since it's a higher-level/more-idiomatically-Objective-C construct than the lower-level/more-C-like `dispatch_async()` call (the two are basically equivalent).